odl/tomo/backends/astra_cuda.py

```diff
@@ -273,6 +241,83 @@ def astra_cuda_back_projector(proj_data, geometry, reco_space, out=None):
+    """
+    # Angular integration weighting factor
+    # angle interval weight by approximate cell volume
+    angle_extent = float(geometry.motion_partition.extent())
+    num_angles = float(geometry.motion_partition.size)
+    scaling_factor = angle_extent / num_angles
+
+    if parse_version(ASTRA_VERSION) < parse_version('1.8rc1'):
+        # Fix inconsistent scaling
+        if isinstance(geometry, Parallel2dGeometry):
+            # Scales with 1 / cell_volume
+            scaling_factor *= float(reco_space.cell_volume)
+        elif isinstance(geometry, FanFlatGeometry):
+            # Scales with 1 / cell_volume
+            scaling_factor *= float(reco_space.cell_volume)
+            # Additional magnification correction
+            src_radius = geometry.src_radius
+            det_radius = geometry.det_radius
+            scaling_factor *= ((src_radius + det_radius) / src_radius)
+        elif isinstance(geometry, Parallel3dAxisGeometry):
+            # Scales with voxel stride
+            # In 1.7, only cubic voxels are supported
+            voxel_stride = reco_space.cell_sides[0]
+            scaling_factor /= float(voxel_stride)
+        elif isinstance(geometry, HelicalConeFlatGeometry):
+            # Scales with 1 / cell_volume
+            # In 1.7, only cubic voxels are supported
+            voxel_stride = reco_space.cell_sides[0]
+            scaling_factor /= float(voxel_stride)
+            # Magnification correction
+            src_radius = geometry.src_radius
+            det_radius = geometry.det_radius
+            scaling_factor *= ((src_radius + det_radius) / src_radius) ** 2
+
+    else:
+        if isinstance(geometry, Parallel2dGeometry):
+            # Scales with 1 / cell_volume
+            scaling_factor *= float(reco_space.cell_volume)
+        elif isinstance(geometry, FanFlatGeometry):
+            # Scales with 1 / cell_volume
+            scaling_factor *= float(reco_space.cell_volume)
+            # Magnification correction
+            src_radius = geometry.src_radius
+            det_radius = geometry.det_radius
+            scaling_factor *= ((src_radius + det_radius) / src_radius)
+        elif isinstance(geometry, Parallel3dAxisGeometry):
+            # Scales with cell volume
+            # currently only square voxels are supported
+            scaling_factor /= reco_space.cell_volume
+        elif isinstance(geometry, HelicalConeFlatGeometry):
+            # Scales with cell volume
+            scaling_factor /= reco_space.cell_volume
+            # Magnification correction
+            src_radius = geometry.src_radius
+            det_radius = geometry.det_radius
+            scaling_factor *= ((src_radius + det_radius) / src_radius) ** 2
+
+            # Correction for scaled 1/r^2 factor in ASTRA's density weighting
+            det_px_area = geometry.det_partition.cell_volume
+            scaling_factor *= (src_radius ** 2 * det_px_area ** 2 /
```

Comment:
I dont get it. You scale by 1/src_radius^2 above and src_radius^2 here. Why not simply remove it?